### PR TITLE
Fix issue of header is getting modified when posting multipart/form-data

### PR DESCRIPTION
**Pull request by sohaniwso2**
Public jira - https://wso2.org/jira/browse/ESBJAVA-5041

### CODE REVIEW

File: modules/kernel/src/org/apache/axis2/transport/http/MultipartFormDataFormatter.java [hunk 1]
```diff
@@ -223,51 +228,70 @@ public String formatSOAPAction(MessageContext messageContext, OMOutputFormat for
...
-
-        }
-        Part[] partsArray = new Part[parts.size()];
-        return parts.toArray(partsArray);
-    }
+	/**
+	 * @param dataOut
+	 * @return
+	 */
+	private Part[] createMultipatFormDataRequest(OMElement dataOut) {
```
Comment (Kishanthan) [inline modules/kernel/src/org/apache/axis2/transport/http/MultipartFormDataFormatter.java]: Looks like this method is not aligned and formatted properly.

File: modules/kernel/src/org/apache/axis2/transport/http/MultipartFormDataFormatter.java [hunk 2]
```diff
@@ -223,51 +228,70 @@ public String formatSOAPAction(MessageContext messageContext, OMOutputFormat for
...
+								part = new FilePart(ele.getQName().getLocalPart(),
+										new ByteArrayPartSource(fileName, ele.getText().getBytes()), contentType, null);
+							}
+						}
+					}
+					if (part == null) {
+						part = new StringPart(ele.getQName().getLocalPart(), ele.getText());
+					}
+				}
+
```
Comment (Kishanthan) [inline modules/kernel/src/org/apache/axis2/transport/http/MultipartFormDataFormatter.java]: Unwanted new line

**Comment by sohaniwso2** [comment]
@Kishanthan : It seems axis2 is using a formatter with the indentation size 2, hence formatted the method accordingly

File: modules/kernel/test/org/apache/axis2/transport/http/MultipartFormDataFormatterTest.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+import org.apache.axiom.soap.SOAPBody;
+import org.apache.axiom.soap.SOAPEnvelope;
+import org.apache.axiom.soap.SOAPFactory;
+import org.apache.axis2.AbstractTestCase;
+import org.apache.axis2.builder.DiskFileDataSource;
+import org.apache.axis2.context.MessageContext;
+import org.apache.commons.fileupload.disk.DiskFileItem;
+import org.apache.commons.fileupload.disk.DiskFileItemFactory;
+
+public class MultipartFormDataFormatterTest extends AbstractTestCase {
```
Comment (Kishanthan) [inline modules/kernel/test/org/apache/axis2/transport/http/MultipartFormDataFormatterTest.java]: Looks like this class is also not formatted according to wso2 style. Please fix it.

Comment (sohaniwso2) [inline modules/kernel/test/org/apache/axis2/transport/http/MultipartFormDataFormatterTest.java]: fixed

File: modules/kernel/test-resources/binaryFile.xml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+   <book id="bk101">
+      <author>Gambardella, Matthew</author>
+      <title>XML Developer's Guide</title>
+      <genre>Computer</genre>
+      <price>44.95</price>
+      <publish_date>2000-10-01</publish_date>
+      <description>An in-depth look at creating applications 
+      with XML.</description>
+   </book>
+ </catalog>
```
Comment (Kishanthan) [inline modules/kernel/test-resources/binaryFile.xml]: EoF new line is missing.

Comment (sohaniwso2) [inline modules/kernel/test-resources/binaryFile.xml]: fixed

File: modules/kernel/test/org/apache/axis2/transport/http/MultipartFormDataFormatterTest.java (hunk 3 above)
Comment (VIKRAMAS) [reply modules/kernel/test/org/apache/axis2/transport/http/MultipartFormDataFormatterTest.java]: Please can you help me how to solve this issue

Comment (VIKRAMAS) [reply modules/kernel/test/org/apache/axis2/transport/http/MultipartFormDataFormatterTest.java]: Please can you help me how to solve this issue

**Pull request merged**